lib/services/identitycenter.go

```diff
@@ -0,0 +1,220 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package services
+
+import (
+	"context"
+
+	"google.golang.org/protobuf/proto"
+
+	identitycenterv1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/identitycenter/v1"
+	"github.com/gravitational/teleport/lib/utils/pagination"
+)
+
+// IdentityCenterAccount wraps a raw identity center record in a new type to
+// allow it to implement the interfaces required for use with the Unified
+// Resource listing.
+//
+// IdentityCenterAccount simply wraps a pointer to the underlying
+// identitycenterv1.Account record, and can be treated as a reference-like type.
+// Copies of an IdentityCenterAccount will point to the same record.
+type IdentityCenterAccount struct {
+	// This wrapper needs to:
+	//  - implement the interfaces required for use with the Unified Resource
+	//    service.
+	//  - expose the existing interfaces & methods on the underlying
+	//    identitycenterv1.Account
+	//  - avoid copying the underlying identitycenterv1.Account due to embedded
+	//    mutexes in the protobuf-generated code
+	//
+	// Given those requirements, storing an embedded pointer seems to be the
+	// least-bad approach.
+
+	*identitycenterv1.Account
+}
+
+// CloneResource creates a deep copy of the underlying account resource
+func (a *IdentityCenterAccount) CloneResource() IdentityCenterAccount {
```

Comment:
Just spotted this one is a pointer receiver and the other one is a value receiver `func (a IdentityCenterAccountAssignment) CloneResource()`